src/tools/rustfmt/src/patterns.rs

```diff
@@ -340,6 +341,7 @@ impl Rewrite for Pat {
                 .map(|inner_pat| format!("({})", inner_pat)),
             PatKind::Err(_) => Err(RewriteError::Unknown),
             PatKind::Deref(_) => Err(RewriteError::Unknown),
+            PatKind::Guard(..) => Err(RewriteError::Unknown),
```

Comment:
It would be better to return the original span unformatted since returning an `Err` will prevent formatting whatever this pattern is in, e.g a match expression.
```suggestion
            PatKind::Guard(..) =>  Ok(context.snippet(self.span).to_owned()),
```

Also, could you please add a test case annotated with `#![feature(guard_patterns)]` in `src/tools/rustfmt/tests/target` so that we can keep track of this on the rustfmt side.

Reply:
I had understood that the rustfmt policy was "don't format new syntax in any way whatsoever when implementing it in rustc, let us handle that". Has this changed/did I misunderstand?

Reply:
When it comes to the policy it's more about implementing formatting rules. We don't consider returning the span unformatted as implementing any formatting.

As for the test case, I think it should be formatted based on what the current expectations are, and I think it's okay if that changes once t-style has a chance to codify rules. Maybe we could just reuse the example from the RFC?

```
match user.subscription_plan() {
    (Plan::Regular if user.credit() >= 100) | (Plan::Premium if user.credit() >= 80) => {
        // Complete the transaction.
    }
    _ => {
        // The user doesn't have enough credit, return an error message.
    }
}
```

Reply:
Sounds good, ty!

Reply:
How does the last commit look?

Reply:
Things look good on the rustfmt side

Reply:
ty!